test/test_dists.jl

```diff
@@ -717,6 +717,14 @@ end
     @test bregman(F, ∇, p, q) ≈ ISdist(p, q)
 end
 
+@testset "CartesianIndex" begin
+    A = reshape(collect(1:9), 3, 3)
+    inds1 = findall(iseven, A)
+    inds2 = findall(isodd, A)
+    @test sum(pairwise(SqEuclidean(), inds1, inds2)) == 52
+    @test euclidean(inds1[1], inds1[1]) === 0.0
+end
```

Comment:
Does this cover a real use case? I find it surprising that somebody would want to do that.

Reply:
There was a feature request in #177. But we could remove it for now and clarify there what the actual intention and the use are.

Reply:
Yes, I'd leave it out until we have a clearer view of why that would be useful. The fact that `CartesianIndex` isn't iterable seems to indicate that it's not intended to be used that way, and it would be absurd to have packages work around that everywhere...